test/test_fork.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2022-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test that pymongo is fork safe."""
+
+import os
+import threading
+from multiprocessing import Pipe
+from test import IntegrationTest, client_context
+from typing import Any, Callable
+from unittest import skipIf
+from unittest.mock import patch
+
+from bson.objectid import ObjectId
+from pymongo import MongoClient
+from pymongo.lock import _ForkLock
+
+
+@client_context.require_connection
+def setUpModule():
+    pass
+
+
+class ForkThread(threading.Thread):
+    def __init__(self, exit_cond: Callable[[], int] = lambda: 0):
+        super().__init__()
+        self.pid: int = -1  # Indicate we haven't started.
+        self.exit_cond = exit_cond
+
+    def run(self):
+        if self.pid < 0:
+            self.pid = os.fork()
+            if self.pid == 0:
+                os._exit(self.exit_cond())
+
+
+class LockWrapper:
+    def __init__(self, fork_thread: ForkThread, lock_type: Any = _ForkLock):
+        self.__lock = lock_type()
+        self.fork_thread = fork_thread
+
+    def __enter__(self):
+        self.__lock.__enter__()
+        if not self.fork_thread.is_alive():
+            self.fork_thread.start()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.__lock.__exit__(exc_type, exc_value, traceback)
+
+    def __getattr__(self, item):
+        return getattr(self.__lock, item)
+
+
+# Not available for versions of Python without "register_at_fork"
+@skipIf(
+    not hasattr(os, "register_at_fork"), "register_at_fork not available in this version of Python"
+)
+class TestFork(IntegrationTest):
+    def setUp(self):
+        self.db = self.client.pymongo_test
+        self.fork_thread = ForkThread()
```

Comment:
`self.fork_thread` should be defined in each test so exit_cond can be passed to the constructor instead of patching exit_cond.